src/storage/write_ahead_log/log_serializer_task.cpp

```diff
@@ -48,13 +49,20 @@ void LogSerializerTask::LogSerializerTaskLoop() {
     std::tie(num_bytes, num_records, num_txns) = Process();
```

Comment:
I've noticed something thing about these metrics, I don't know if it's good or bad but I just wanted to point it out. `Process()` Has its own loop, which continuously loops as long as there are more log records to serialize. That means that `Process()` doesn't return until there's a break in log records. The two consequences of this are

1. If there's a steady stream of logs then you can end up with a very long interval that contains a lot of log records.
2. While there's a steady stream of logs then you won't get any metrics until there's a break in the logs.

So if you're running a workload with a lot of small transactions, you may end up with a single interval that contains the metrics of the entire workload.

If you wanted more granular metrics it might be a good idea to push all the metrics stuff into `Process()` itself.